eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java

```diff
@@ -101,14 +99,14 @@ private void checkNoExternalDependencyExposed(DetailAST methodDefToken) {
         final DetailAST typeToken = methodDefToken.findFirstToken(TokenTypes.TYPE);
         if (typeToken != null) {
             getInvalidReturnTypes(typeToken).forEach(
-                (token, returnTypeName) -> log(token, String.format(EXTERNAL_DEPENDENCY_ERROR, returnTypeName)));
+                (token, returnTypeName) -> log(token, String.format("Class ''%s'', is a class from external dependency. You should not use it as a return type.", returnTypeName)));
         }
 
         // Checks for the parameters of the method
         final DetailAST parametersToken = methodDefToken.findFirstToken(TokenTypes.PARAMETERS);
         if (parametersToken != null) {
             getInvalidParameterTypes(parametersToken).forEach(
-                (token, returnTypeName) -> log(token, String.format(EXTERNAL_DEPENDENCY_ERROR, returnTypeName)));
+                (token, parameterTypeName) -> log(token, String.format("Class ''%s'', is a class from external dependency. You should not use it as a method argument type.", parameterTypeName)));
```

Comment:
IMO, it is better to keep the `private static string label` for the error message, having strings like this is not good for maintenance.
What about having 3 static strings
```
private static String EXTERNAL_DEPEPENDENCY_ERROR_TITLE = "Class ''%s'', is a class from external dependency. . You should not use it as a";
private static String EXTERNAL_DEPEPENDENCY_ERROR_RETURN_TYPE = "return type";
private static String EXTERNAL_DEPEPENDENCY_ERROR_ARG_TYPE = "argument type";
``` 

then we can have a method like `getErrorMessage( EXTERNAL_DEPEPENDENCY_ERROR_ARG_TYPE)` to use the error title with what ever ending string we need.

Reply:
Agree on putting common string out as private static final String. For the  "return type" and "argument type", because they are only used once. It should not necessary to add those two fields.